docs/configuration/jobs.rst

```diff
@@ -173,6 +173,145 @@ Note that you will have to trigger replication manually using the ``zrepl signal
+
+The ``stop_after_error`` parameter specifies whether the snapshot process will stop after a failed hook. The default
+is ``false``, meaning subsequent scripts will run and the snapshot will be taken even if a pre-snapshot hook process exits with an error or the timeout is exceeded.
+Setting ``stop_after_error: true`` will prevent snapshots from being taken if the pre-snapshot hook fails, preventing
+inconsistent snapshots if, for example, the pre-snapshot script is asking a database to write its state to disk. ``stop_after_error: true`` will skip any scripts after the script with an error, both pre- and post-hooks.
+
+The script's standard output is logged at level INFO by zrepl. The standard error is logged at level WARN.
+
+Examples:
+
+.. code-block:: bash
+    :caption: postgresql-pre-snapshot.sh
+
+    #!/bin/sh
+
+    ###
+    # Settings
+    ###
+    PSQL_TIMEOUT_MINUTES=10
+
+    [ "$ZREPL_DRYRUN" = "true" ] && exit
+
+    /usr/local/bin/psql \
+      --echo-errors \
+      --no-password \
+      --no-psqlrc \
+      -U postgres \
+      -c "SET statement_timeout TO $(($PSQL_TIMEOUT_MINUTES * 60)); CHECKPOINT;"
+    # Exit with psql's exit code
+    exit $?
+
+The following only works with InnoDB tables, and you MUST run the mysql-post-snapshot.sh script as a post-hook or the pre-snapshot script will hold a read lock for MYSQL_LOCK_TIMEOUT_SECONDS.
+
+.. code-block:: bash
+    :caption: mysql-pre-snapshot.sh
+
+    #!/bin/sh
+
+    ###
+    # Settings
+    ###
+    MYSQL_LOCK_TIMEOUT_SECONDS=30
+    MYSQL_LOCK_FILE=/var/run/mysqld/zrepl.lock
+
+    ###
+    # You must either create a /root/.my.cnf file
+    # with credentials or add -uUSERNAME -pPASSWORD
+    # below. Recommend creating a user that has only
+    # the LOCK TABLES and RELOAD priviledge to avoid
+    # exposing more than necessary to this script:
+    # https://dev.mysql.com/doc/refman/5.7/en/flush.html#flush-tables-with-read-lock-with-list
+    #
+    # DO NOT USE IN A DOCKER CONTAINER WITH NO INIT;
+    # the background mysql client with be an unreaped zombie:
+    # https://stackoverflow.com/questions/49162358/docker-init-zombies-why-does-it-matter
+    ###
+    mysql -NB <<-EOF > /dev/null &
+      FLUSH TABLES WITH READ LOCK;
+      DELIMITER ;;
+      SYSTEM touch "$MYSQL_LOCK_FILE"
+      SYSTEM sh -c 'timeout=$MYSQL_LOCK_TIMEOUT_SECONDS; elapsed=0; while test -e "$MYSQL_LOCK_FILE" -a \$elapsed -lt \$timeout; do sleep 1; elapsed=\$((\$elapsed + 1)); done; test \$elapsed -ge \$timeout && rm -f "$MYSQL_LOCK_FILE";'
+      EXIT
+    EOF
+
+    # Wait for the lock to be created by the preceding command
+    elapsed=0
+    while [ ! \( -e "$MYSQL_LOCK_FILE" \) -a $elapsed -lt $MYSQL_LOCK_TIMEOUT_SECONDS ]; do
+      sleep 1
+      elapsed=$(($elapsed + 1))
+    done
+
+    if [ $elapsed -ge $MYSQL_LOCK_TIMEOUT_SECONDS ]; then
+      2> echo "Couldn't acquire MySQL READ LOCK"
+      exit 1
+    fi
+
+.. code-block:: bash
+    :caption: mysql-post-snapshot.sh
```

Comment:
I think pre and post code belong into the same file.
Of course that's a matter of taste, but in particular, you have duplicate definitions of setup-specific information like `MYSQL_LOCK_FILE` in both files, which calls for trouble (admin will only change one file => pre and post use different locks => things hang until timeout => unhappy admin)